nix/mobile/android/build.nix

```diff
@@ -145,15 +147,15 @@ in stdenv.mkDerivation rec {
   buildPhase = let
     adhocEnvVars = optionalString stdenv.isLinux
       "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${makeLibraryPath [ pkgs.zlib ]}";
+    gradleFlags = [
+      "--console=plain"
+      "-Dmaven.repo.local='${deps.gradle}'"
+    ]
+    ++ lib.optionals (buildType != "debug") [ "--offline" "--no-build-cache" "--no-daemon" "--no-scan" "--no-watch-fs" ]
+    ++ lib.optionals (buildType == "debug") [ "--build-cache" "--parallel" ];
```

Comment:
This makes no sense. We should be always using these flags:
```
--offline
--no-build-cache
--no-daemon
--no-scan
--no-watch-fs
```
Using the opposite makes absolutely no sense in Nix context, especially not `--build-cache`.

Please remember that every nix build is built in a fresh sandbox environment that only has the source provided from the repo, and nothing else. There is no internet and there is no persistent storage.

Even if you enable cache it will have NO EFFECT since there is no persistent storage between builds.

Reply:
I agree, I tested without this change and build time average was similar.
I guess we could add `--parallel` to the existing flags list since that may speed up builds.

Reply:
https://docs.gradle.org/current/userguide/performance.html#parallel_execution

Reply:
we could also leave remove the  no-daemon flag 
ref -> https://docs.gradle.org/current/userguide/performance.html#enable_daemon

> On developer machines, you should see a significant performance improvement. On CI machines, long-lived agents benefit from the Daemon. But short-lived machines don’t benefit much. Daemons automatically shut down on memory pressure in Gradle 3.0 and above, so it’s always safe to leave the Daemon enabled.

Reply:
No, that also doesn't make sense. Learn about what a Nix sandox is.

Reply:
Parallel flag probably makes sense tho.

Reply:
Thanks!